src/merge.py

```diff
@@ -13,10 +13,17 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     """
     i, j = 0, 0
     z = []  # a new list to copy elements into
-    # FIXME: fill out the loop so you merge the lists
-    # until one of them is empty
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
-    # At least one of the lists is empty now. Copy the
-    # remainder of the other into z.
-    return z
+        if x[i] < y[j]:
+            z.append(x[i])
+            i += 1
+        else:
+            z.append(y[j])
+            j+=1
+    if i >= len(x):
+        z.extend(y[j:])
+        return z
+    else:
+        z.extend(x[i:])
+        return z
```

Comment:
This is slightly more complicated than it needs to be. You don't need to check which list is now empty, because you will want to add both `x[i:]` and `y[j:]` anyway. At least one of them is empty, but if you `extend` with an empty list, it just doesn't do anything, so that is entirely safe. You could, therefore, change your code to

```python
z.extend(x[i:])
z.extend(y[j:])
return z
```

It might look like we are adding two lists to `z` (because we are) but the empty one doesn't add any elements.